src/array.rs

```diff
@@ -30,67 +32,69 @@ use crate::error::{
-/// [`from_slice`](#method.from_slice) or [`from_array`](#from_array).
+/// - Allocated by NumPy: Constructed via other methods, like [`ToPyArray`] or
+/// [`from_slice`][Self::from_slice] or [`from_array`][Self::from_array].
 ///
-/// These methods allocate memory in Python's private heap.
+/// These methods allocate memory in Python's private heap via NumPy's API.
 ///
-/// In both cases, **PyArray is managed by Python GC.**
-/// So you can neither retrieve it nor deallocate it manually.
+/// In both cases, `PyArray` is managed by Python so it can neither be moved from
+/// nor deallocated manually.
 ///
-/// # Reference
-/// Like [`new`](#method.new), all constractor methods of `PyArray` returns `&PyArray`.
+/// # References
 ///
-/// This design follows
-/// [pyo3's ownership concept](https://pyo3.rs/main/doc/pyo3/index.html#ownership-and-lifetimes).
+/// Like [`new`][Self::new], all constructor methods of `PyArray` return a shared reference `&PyArray`
+/// instead of an owned value. This design follows [PyO3's ownership concept][pyo3-memory],
+/// i.e. the return value is GIL-bound owning reference into Python's heap.
 ///
+/// # Element type and dimensionality
 ///
-/// # Data type and Dimension
-/// `PyArray` has 2 type parametes `T` and `D`. `T` represents its data type like
-/// [`f32`](https://doc.rust-lang.org/std/primitive.f32.html), and `D` represents its dimension.
+/// `PyArray` has two type parametes `T` and `D`.
+/// `T` represents the type of its elements, e.g. [`f32`] or [`PyObject`].
+/// `D` represents its dimensionality, e.g [`Ix2`][type@Ix2] or [`IxDyn`][type@IxDyn].
 ///
-/// All data types you can use implements [Element](../types/trait.Element.html).
+/// Element types are Rust types which implement the [`Element`] trait.
+/// Dimensions are represented by the [`ndarray::Dimension`] trait.
 ///
-/// Dimensions are represented by ndarray's
-/// [Dimension](https://docs.rs/ndarray/latest/ndarray/trait.Dimension.html) trait.
+/// Typically, `Ix1, Ix2, ...` are used for fixed dimensionality arrays,
+/// and `IxDyn` is used for dynamic dimensionality arrays. Type aliases
+/// for combining `PyArray` with these types are provided, e.g. [`PyArray1`] or [`PyArrayDyn`].
 ///
-/// Typically, you can use `Ix1, Ix2, ..` for fixed size arrays, and use `IxDyn` for dynamic
-/// dimensioned arrays. They're re-exported from `ndarray` crate.
-///
-/// You can also use various type aliases we provide, like [`PyArray1`](./type.PyArray1.html)
-/// or [`PyArrayDyn`](./type.PyArrayDyn.html).
-///
-/// To specify concrete dimension like `3×4×5`, you can use types which implements ndarray's
-/// [`IntoDimension`](https://docs.rs/ndarray/latest/ndarray/dimension/conversion/trait.IntoDimension.html)
-/// trait. Typically, you can use array(e.g. `[3, 4, 5]`) or tuple(e.g. `(3, 4, 5)`) as a dimension.
+/// To specify concrete dimension like `3×4×5`, types which implement the [`ndarray::IntoDimension`]
```

Comment:
I didn't know that this works on docs.rs.... maybe my knowledge also needs to be updated.

Reply:
Documentation is always built together with all the documentation of the dependencies so that one can link into it. With intra-doc links everything that is within scope can be linked to, including dependencies. You can find an example where this is already used in https://docs.rs/numpy/0.16.2/numpy/array/struct.PyArray.html#method.from_owned_array